### PR TITLE
add support for more execution modes

### PR DESCRIPTION
**Pull request by jjno91**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -41,44 +45,87 @@ def create_backend(bucket: str, key: str, region: str = 'us-east-2'):
     f.write('terraform {\n')
     f.write('  backend "s3" {\n')
     f.write('    bucket = "' + bucket + '"\n')
-    f.write('    key    = "' + key + '.json"\n')
+    f.write('    key    = "' + key + '"\n')
```
Comment (reeves122) [inline main.py:48]: Just FYI you can write multiline strings using:
```
f.write('''
terraform {\n
  backend "s3" {\n
  ...
''')
```

File: main.py [hunk 2]
```diff
@@ -41,44 +45,87 @@ def create_backend(bucket: str, key: str, region: str = 'us-east-2'):
...
+    repo: str = get_repo_name()
     branch: str = Repository('.').head.shorthand
-    key: str = repo + '/' + branch
+    key: str = repo + '/' + branch + '.json'  # .json helps with manually manipluating state files in S3
+    os.environ["TF_INPUT"] = "false"
+    os.environ["TF_IN_AUTOMATION"] = "true"
+    os.environ["TF_VAR_name"] = repo
+    os.environ["TF_VAR_branch"] = branch
 
     if bucket == '':
```
Comment (reeves122) [inline main.py:122]: FYI this can be shortened to simply `if not bucket`

File: test_main.py [hunk 3]
```diff
@@ -17,10 +17,15 @@ def setup():
...
-    os.remove('terraform.tfplan')
-    os.remove('backend.tf')
+
+    # noinspection PyBroadException
+    try:
+        shutil.rmtree(test_directory)
+        shutil.rmtree('.terraform')
+        os.remove('terraform.tfplan')
+        os.remove('backend.tf')
+    except Exception:
```
Comment (reeves122) [inline test_main.py:27]: Can be fixed later but its generally code-smell to use blanket exceptions in Python

**Review by reeves122 — APPROVED** [review]

File: main.py (hunk 1 above)
Comment (jjno91) [reply main.py:48]: i couldn't get variable injection to work in the multiline string for lines 47-49

File: main.py (hunk 2 above)
Comment (jjno91) [reply main.py:122]: added to my working copy

**Pull request merged**